internal/model/model.go

```diff
@@ -84,8 +84,12 @@ type service interface {
 }
 
 type Model struct {
-	cfg             *config.Wrapper
-	db              *leveldb.DB
+	cfg *config.Wrapper
+	db  *leveldb.DB
+
+	cache    *db.NamespacedKV
+	cacheMut sync.Mutex
```

Comment:
I guess it's there for a reason. There might bemore reads than writes hence RWMutex would be beneficial.